### PR TITLE
Build conversation structure with ConvoKit

### PR DESCRIPTION
**Pull request by sophieball**
#63 

This is just creating the structure. I tried to add a Bazel test but matplotlib is not happy with it and gives me `Gdk-CRITICAL **: 03:53:25.062: gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed`. I didn't explicitly import `matplotlib` in `conversation_construct.py`; it may be imported by convokit

### CODE REVIEW

File: src/conversation_struct.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  for convo in corpus.iter_conversations():
+    convo_id = convo.get_id()
+    if convo_id in conversation_label:
+      convo.meta["label"] = conversation_label[convo_id]
+  return corpus
+
+
+if __name__ == "__main__":
+  comments = pd.read_csv("src/data/pr_comments.csv")
+  body = pd.read_csv("src/data/pr_body.csv")
```
Comment (CaptainEmerson) [inline src/conversation_struct.py:255]: ditto for this file being removed

File: main/train_prompt_types.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+import pickle
+
+from src import conversation_struct
+from src import predict_bad_conver_helpers as hp
+
+VERBOSITY = 10000
+N_TYPES = 6
+
+# read in data
```
[the remark this answers is not in the document]
Comment (sophieball) [reply main/train_prompt_types.py:78]: Done.

File: src/predict_bad_conver_helpers.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
+# Lint as: python3
+"""TODO(hsqq): DO NOT SUBMIT without one-line documentation for predict_bad_convo_helpers.
```
[the remark this answers is not in the document]
Comment (sophieball) [reply src/predict_bad_conver_helpers.py:2]: No... I thought Google somehow would hahahaha! I'll write some description there

File: src/conversation_struct.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+  # Update conversation labels
+  for convo in corpus.iter_conversations():
+    convo_id = convo.get_id()
+    if convo_id in conversation_label:
+      convo.meta["label"] = conversation_label[convo_id]
+  return corpus
+
+
+if __name__ == "__main__":
+  comments = pd.read_csv("src/data/pr_comments.csv")
```
[the remark this answers is not in the document]
Comment (sophieball) [reply src/conversation_struct.py]: you're right

File: src/conversation_struct.py (hunk 1 above)
Comment (sophieball) [reply src/conversation_struct.py]: done

File: src/predict_bad_conver_helpers.py [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+  if "prompt_types" in feature_list:
+    features_to_use += prompt_type_names
+  if "politeness_strategies" in feature_list:
+    features_to_use += politeness_names
+
+  feature_subset = feature_table[features_to_use]
+
+  #c0_feats = feature_subset.loc[labeled_pairs_df.c0].values
+  #c1_feats = feature_subset.loc[labeled_pairs_df.c1].values
+  #c_feats = feature_subset.loc[feature["conversation_id"]]
```
[the remark this answers is not in the document]
Comment (sophieball) [reply src/predict_bad_conver_helpers.py]: done

**Comment by CaptainEmerson** [comment]
```
Traceback (most recent call last):
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/__main__/main/train_prompt_types.py", line 119, in <module>
    train_prompt(corpus, "pt_model_10K.files")
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/__main__/main/train_prompt_types.py", line 102, in train_prompt
    pt.fit(corpus)
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__convokit_2_3_2_4/convokit/prompt_types/promptTypeWrapper.py", line 111, in fit
    self.pipe.fit(corpus)
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__scikit_learn_0_23_1/sklearn/pipeline.py", line 335, in fit
    self._final_estimator.fit(Xt, y, **fit_params_last_step)
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__convokit_2_3_2_4/convokit/prompt_types/promptTypes.py", line 129, in fit
    self.prompt_embedding_model = fit_prompt_embedding_model(prompt_input, ref_input,
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__convokit_2_3_2_4/convokit/prompt_types/promptTypes.py", line 461, in fit_prompt_embedding_model
    ref_vect = ref_tfidf_model.fit_transform(ref_input)
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__scikit_learn_0_23_1/sklearn/feature_extraction/text.py", line 1840, in fit_transform
    X = super().fit_transform(raw_documents)
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__scikit_learn_0_23_1/sklearn/feature_extraction/text.py", line 1198, in fit_transform
    vocabulary, X = self._count_vocab(raw_documents,
  File "/usr/local/google/home/emersonm/toxicity-detector/bazel-bin/main/train_prompt_types.runfiles/deps_pypi__scikit_learn_0_23_1/sklearn/feature_extraction/text.py", line 1129, in _count_vocab
    raise ValueError("empty vocabulary; perhaps the documents only"
ValueError: empty vocabulary; perhaps the documents only contain stop words

```

**Comment by sophieball** [comment]
- There's a variable controlling for G/GH. But you don't need to worry about it - the burden is on my side. When I run it I'll add another command line argument then it will run it as GH
- It doesn't dump the corpus data now, because it means writing G's text to disk. 
- It now prints the structure of the first conversation to the log, for sanity checking
- The clustering output will be saved in `bad_conver.log`, and the trained model is in the directory `pt_model_10K.files`, both in the bazel-bin runfiles directory

**Comment by CaptainEmerson** [comment]
Looks like the build is failing, though.

**Pull request merged**